crates/hyperswitch_domain_models/src/payment_methods.rs

```
@@ -138,6 +138,68 @@ impl PaymentMethod {
     pub fn get_payment_method_subtype(&self) -> Option<storage_enums::PaymentMethodType> {
         self.payment_method_subtype
     }
+
+    #[cfg(all(
+        any(feature = "v1", feature = "v2"),
+        not(feature = "payment_methods_v2")
+    ))]
+    pub fn get_common_mandate_reference(
+        &self,
+    ) -> Result<diesel_models::CommonMandateReference, ParsingError> {
+        let payments_data = self
+            .connector_mandate_details
+            .clone()
+            .map(|mut mandate_details| {
+                mandate_details
+                    .as_object_mut()
+                    .map(|obj| obj.remove("payouts"));
+
+                serde_json::from_value::<diesel_models::PaymentsMandateReference>(mandate_details)
+                    .inspect_err(|err| {
+                        router_env::logger::error!("Failed to parse payments data: {}", err);
+                    })
+            })
+            .transpose()
+            .map_err(|_| ParsingError::StructParseFailure("Failed to parse payments data"))?;
```

Avoid discarding error, log the error.